src/containers/sb-file-uploader.jsx

```diff
@@ -79,10 +79,13 @@ class SBFileUploader extends React.Component {
             reader.readAsArrayBuffer(thisFileInput.files[0]);
             // extract the title from the file and set it as current project title
             if (thisFileInput.files[0].name) {
-                const matches = thisFileInput.files[0].name.match(/^(.*)\.sb3$/);
+                const matches = thisFileInput.files[0].name.match(/^(.*?)(\.sb[23]?)?$/);
```

Comment:
To me this reads "Any sequence of characters _optionally_ followed by `.sb`, `.sb2` or `.sb3`".  The file picker we're creating only allows you to pick `.sb2` or `.sb3` files, so I think we could require one of these extensions: `/^(.*)\.sb[23]$/`. Curious why the change to a more expansive set of filenames?

Reply:
Here is my reasoning:

If, after some future changes to the code outside of this line, this line runs, and gets a filename like

`Why I sing.sb`
or
`Why I sing`

should it just decide to completely discard "Why I sing" and not change the project title to that? That seems like it would be a mistake.

Reply:
We should update this when we change the code then — what if the code never changes like that? Then this is set up for a situation that never occurs, and confuses the intent. As it is now, it looks like there's possibly a bug in the file input because it's only allowing .sb2 or .sb3 files, but this part of the code can allow .sb files as well. So which one is right? Without context would need to investigate. If we're consistent then there's no ambiguity.

Reply:
I think we have a philosophical difference here, but it's a place that doesn't matter much, and I'm happy to do it your way.